### PR TITLE
Feature/formater/html

### PR DESCRIPTION
**Pull request by webit4me**
Here is a functioning prototype for the HTML formatter.
please note, at the moment it relies on the 3rd party front-end libraries (online) which i am aiming to embed them as soon as i get the chance

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 behat.yml
 vendor
 composer.lock
+.idea
```
Comment (WouterJ) [inline .gitignore]: you should add this to your global gitignore or the not commited git ignore (`.git/info/exclude`). It should not be part of the project's gitignore.

File: src/Behat/Behat/Output/Node/Printer/Html/templates/error.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<div class="alert alert-danger stepError" role="alert">
+    <span class="glyphicon glyphicon-exclamation-sign" aria-hidden="true"></span>
+    <span class="sr-only">Error:</span>
+    <pre>{error}</pre>
+</div>
```
Comment (WouterJ) [inline src/Behat/Behat/Output/Node/Printer/Html/templates/error.html]: all files should end with a line feed character

File: src/Behat/Behat/Output/Node/Printer/Html/HtmlPrinterTrait.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+namespace Behat\Behat\Output\Node\Printer\Html;
+
+use Behat\Testwork\Output\Printer\OutputPrinter;
+
+/**
+ * To retrieve (html printer) helper .
+ *
+ * @author Ali Bahman <abn@webit4.me>
+ */
+trait HtmlPrinterTrait
```
Comment (WouterJ) [inline src/Behat/Behat/Output/Node/Printer/Html/HtmlPrinterTrait.php]: Behat can't use traits, as it needs to be compatible with PHP 5.3

**Comment by WouterJ** [comment]
You should add scenario's for this feature.

Also, I don't like the way the output printers are set. This has to be refactored once #676 is merged, as it introduces a Filesystem output printer.

File: src/Behat/Behat/Output/Node/Printer/Html/templates/error.html (hunk 2 above)
Comment (webit4me) [inline src/Behat/Behat/Output/Node/Printer/Html/templates/error.html]: I made sure all templates having a line feed at the end

**Comment by stof** [comment]
@webit4me can you rebase your PR so that it does not include unrelated commits ?

**Comment by webit4me** [comment]
@stof, rebased as per your request.

**Comment by stof** [comment]
you still have commits from @WouterJ which don't seem related to this PR

**Comment by webit4me** [comment]
ah, then i might got it wrong. I thought you want me to fixup (like squash) my latest commits on this branch (which were around WouterJ correction/improvement suggestion) and only keep the first main commit.
Going to lookup up to learn how to the what you are asking correctly :)

**Comment by WouterJ** [comment]
What @stof wants you to do (given that `origin` points to this repository and `webit4me` to your fork):

    $ git fetch origin
    $ git checkout feature/formatter/html
    $ git rebase origin/master
    # ... fix any conflicts (if any) and run git rebase --continue after it
    $ git push webit4me -f feature/formatter/html

**Comment by webit4me** [comment]
@WouterJ, thanks a lot :) , but one more question to be clear when you are saying "origin point to the repository" by repository you mean the main "Behat" repo? since at the moment my origin is pointing to the forked version on my webit4me 

$ git remote -v
origin	https://github.com/webit4me/Behat.git (fetch)
origin	https://github.com/webit4me/Behat.git (push)

should i point my origin to the https://github.com/behat/Behat.git  then follow your steps ?

**Comment by WouterJ** [comment]
@webit4me there are 2 standards, I always love to call the original repository origin and my fork the same as my username (this is also the convention recommended by github). Some other people call their own repository origin and the original repository upstream.

Any way, you have to add this as a remote:

    $ git remote add upstream https://github.com/Behat/Behat.git

Then you can optionally rename the remotes to be equal to what I use:

    $ git remote rename origin webit4me
    $ git remote rename upstream origin

**Comment by webit4me** [comment]
Thank you very much for teaching me this :)  
I'm going to tidy things up and get back to you soon
Cheers

**Comment by webit4me** [comment]
Just realised that i've spelled the branch name incorrectly as well :( , if its alright im going to create new branch and open a new request

**Pull request closed without merging**